pkg/deploy/dashboard/deployment_dashboard.go

```diff
@@ -22,7 +23,43 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+const CHE_SELF_SIGNED_MOUNT_PATH = "/public-certs/che-self-signed"
+const CHE_CUSTOM_CERTS_MOUNT_PATH = "/public-certs/custom"
+
 func (d *Dashboard) getDashboardDeploymentSpec() (*appsv1.Deployment, error) {
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	var envVars []corev1.EnvVar
+
+	volumes, volumeMounts = d.provisionCustomPublicCA(volumes, volumeMounts)
+
+	selfSignedCertUsed, err := deploy.IsSelfSignedCertificateUsed(d.deployContext)
```

Comment:
`deploy.IsSelfSignedCertificateUsed` is it pretty heavy check.
For your case just check existence `self-signed-certificated` secret

Reply:
well, I would agree with you but I can't:
1. `IsSelfSignedCertificateUsed` first of all check is secret exists and if it does.
2. it's exactly the same logic that is used for Che Server, so it suggested to me that checking is SA secret exists is not enough.

But then I finally agree. Let me refactor it a bit and ask you take a look a new commit.

Reply:
here we go https://github.com/eclipse-che/che-operator/pull/1073/commits/cfe5333badbd017b07fe9b64248d613abf6ce7ce